heft-plugins/heft-jest-plugin/src/JestRealPathPatch.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'node:path';
+import { realNodeModulePath } from '@rushstack/real-node-module-path';
+
+const jestResolvePackageFolder: string = path.dirname(require.resolve('jest-resolve/package.json'));
+const jestResolveFileWalkersPath: string = path.resolve(jestResolvePackageFolder, './build/fileWalkers.js');
+
+const jestUtilPackageFolder: string = path.dirname(
+  require.resolve('jest-util/package.json', { paths: [jestResolvePackageFolder] })
+);
+const jestUtilTryRealpathPath: string = path.resolve(jestUtilPackageFolder, './build/tryRealpath.js');
+
+const fileWalkersModule: {
+  realpathSync: (filePath: string) => string;
+} = require(jestResolveFileWalkersPath);
+fileWalkersModule.realpathSync = realNodeModulePath;
+
+const tryRealpathModule: {
+  default: (filePath: string) => string;
+} = require(jestUtilTryRealpathPath);
+tryRealpathModule.default = (input: string): string => {
+  try {
+    return realNodeModulePath(input);
+  } catch (error) {
+    if (error.code !== 'ENOENT' && error.code !== 'EISDIR') {
```

Comment:
Use utility functions from node-core-library?

Reply:
Avoiding here because this loads in every jest worker, and even the FileSystem module itself is pretty heavy.